server/services/group.js

```diff
@@ -77,8 +77,8 @@ function removeUser(group, userId) {
 
 function makeJoinRequest(groupName, userId) {
   return userService.exists(userId)
-    .then(() => JoiHelper.validate(userId, ID_SCHEMA))
-    .then(id => Group.makeJoinRequest(groupName, id));
+    .then(() => Promise.all([JoiHelper.validate(userId, ID_SCHEMA), getGroup(groupName)]))
```

Comment:
Weird nit: can you change the order of these promises so the next line uses the results in order? I think it'd be just a tiny bit more readable